codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostRequestDecoderTest.java

```diff
@@ -976,4 +979,97 @@ public void testDecodeMultipartRequest() {
             assertTrue(req.release());
         }
     }
+
+    @Test
+    public void testBIgFileUploadDecoder() throws IOException {
+        int fileSize = 100000000; // set Xmx to a number lower than this and it crashes
+        int bytesPerChunk = 1000000;
+
+        String prefix = "--861fbeab-cd20-470c-9609-d40a0f704466\n" +
+                        "Content-Disposition: form-data; name=\"image\"; filename=\"guangzhou.jpeg\"\n" +
+                        "Content-Type: image/jpeg\n" +
+                        "Content-Length: " + fileSize + "\n" +
+                        "\n";
+
+        String suffix = "\n" +
+                        "--861fbeab-cd20-470c-9609-d40a0f704466--\n";
+
+        HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/upload");
+        request.headers().set("content-type", "multipart/form-data; boundary=861fbeab-cd20-470c-9609-d40a0f704466");
+        request.headers().set("content-length", prefix.length() + fileSize + suffix.length());
+
+        // First Test using Disk
+        HttpDataFactory factory = new DefaultHttpDataFactory(true);
+
+        HttpPostMultipartRequestDecoder decoder = new HttpPostMultipartRequestDecoder(factory, request);
+        decoder.offer(new DefaultHttpContent(Unpooled.wrappedBuffer(prefix.getBytes(CharsetUtil.UTF_8))));
+
+        byte[] body = new byte[bytesPerChunk];
+        Arrays.fill(body, (byte) 1);
+        for (int i = 0; i < fileSize / bytesPerChunk; i++) {
+            ByteBuf content = Unpooled.wrappedBuffer(body, 0, bytesPerChunk);
+            decoder.offer(new DefaultHttpContent(content)); // **OutOfMemory here**
+            content.release();
+        }
+
+        decoder.offer(new DefaultHttpContent(Unpooled.wrappedBuffer(suffix.getBytes(CharsetUtil.UTF_8))));
+        decoder.offer(new DefaultLastHttpContent());
+        FileUpload data = (FileUpload) decoder.getBodyHttpDatas().get(0);
+        assertEquals(data.length(), fileSize);
+        // To not be done since will load full file on memory: assertEquals(data.get().length, fileSize);
+        Field privateByteBufField;
+        ByteBuf field = null;
+        try {
+            privateByteBufField = HttpPostMultipartRequestDecoder.class.getDeclaredField("undecodedChunk");
```

Comment:
Can we change the test to not depend on reflection ?

Reply:
We have to have public access in any way to this private field, or at least the result of `capacity()` method, since this is the only way to check that used memory is below a certain amount.
One way I think of could be:
- make an additional public method that returns the current amount of memory used by this buffer (`undecodedChunk.capacity()`)

However, the issue might be that this method seems to have no interest outside of this kind of tests, but the advantages are that no reflection is needed and that this information is harmless (giving only access to the capacity result, not the buffer itself).

@normanmaurer WDYT ?

Reply:
If there is no other way around make the method package-private and call it from the test. Also add some javadocs that this method is only package-private because of the test.

Reply:
OK, I will follow your idea. It's better to have this method package private indeed!